### ISSUE TITLE
Wrong type infered for struct with default type parameter

### ISSUE THREAD

**Issue by jrmuizel** [intellij-rust/intellij-rust#3999]
Given the following code

```rust
struct B;

struct A<T = B> {
    data: T
}

fn foo() -> A<i32> {
    A { data: 4 }
}
```
intellij-rust gives an error on the '4' saying 'Expected `B` found `i32`'. rustc does not give an error.
